util/clusterauth/clusterauth.go

```diff
@@ -165,19 +173,25 @@ func InstallClusterManagerRBAC(clientset kubernetes.Interface, ns string) (strin
 		if len(serviceAccount.Secrets) == 0 {
 			return false, nil
 		}
-		secretName = serviceAccount.Secrets[0].Name
-		return true, nil
+		// Scan all secrets looking for one of the correct type:
+		for _, oRef := range serviceAccount.Secrets {
+			var getErr error
+			secret, getErr = clientset.CoreV1().Secrets(ns).Get(oRef.Name, metav1.GetOptions{})
```

Comment:
minor - can you rename to `err`?

Reply:
This results in a change from 1 to N+1, would you typically expect this to return on the 1st secret? There's no way to get this from `oRef`?

Reply:
Was a little nervous about using "err" here as it's declared outside the wrap call, but it looks like it's probably safe to do so, will adjust.

Reply:
I don't think there is any way to get it from oRef unfortunately, the type is an attribute specific to secret and thus not present on the generic ObjectReference. However in this case it is unlikely a SA has more than 3 secrets, and I can't imagine any scenario we'd see more than 5-10.

It would often return on the 1st secret I'm sure given no one else has found this problem yet.